homeassistant/components/accuweather/weather.py

```diff
@@ -177,9 +170,7 @@ def forecast(self) -> list[Forecast] | None:
                 ],
                 ATTR_FORECAST_UV_INDEX: item["UVIndex"][ATTR_VALUE],
                 ATTR_FORECAST_WIND_BEARING: item["WindDay"][ATTR_DIRECTION]["Degrees"],
-                ATTR_FORECAST_CONDITION: [
-                    k for k, v in CONDITION_CLASSES.items() if item["IconDay"] in v
-                ][0],
+                ATTR_FORECAST_CONDITION: CONDITION_MAP.get(item["IconDay"]),
```

Comment:
This would previously hard fail, now the condition will instead be set to `None`

Reply:
Good catch, thanks!